main/src/cgeo/geocaching/utils/UriUtils.java

```diff
@@ -25,27 +42,131 @@ private UriUtils() {
+     * * some special volume mappings e.g. for "home" (points to "Document" directory)
+     * * A dummy entry for unit-tests
+     *
+     * In an Uri, the "volume id" is typically written directly behind "/tree". In the following uri
+     * the "volumeId" is "primary" (the most common one, refering to internal storage):
+     * content://com.android.externalstorage.documents/tree/primary%3Acgeo
+     */
+    @NonNull
+    private static Map<String, String> getVolumeMap() {
+        final Map<String, String> volumeMap = new HashMap<>();
+
+        //add a fake volume entry for unit tests. This entry has an uuid which will not be used in real-life
+        volumeMap.put(UNITTEST_VOLUME_ID, UNITTEST_VOLUME_NAME);
+
+        //add special volume ids that we know of
+        volumeMap.put("primary", null); //the most common one where we will NOT put text for
+        volumeMap.put("home", "[Documents]"); //example Uri pointing to /Documents/cgeo: content://com.android.externalstorage.documents/tree/home%3Acgeo
+
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && CgeoApplication.getInstance() != null) {
+            final Context context = CgeoApplication.getInstance().getApplicationContext();
+            final StorageManager storageManager = ContextCompat.getSystemService(context, StorageManager.class);
+            final List<StorageVolume> storageVolumes = storageManager.getStorageVolumes();
+            for (StorageVolume sv : storageVolumes) {
+                volumeMap.put(sv.getUuid(), sv.getDescription(context));
+            }
+        } else {
+            //at least we create a default entry for the primary drive
+            volumeMap.put(null, "internal shared storage");
+        }
+        return volumeMap;
+    }
+
+//    //@RequiresApi(24)
+//    private String getDisplayPath(@NonNull Context context, @NonNull Uri uri) {
+//        ContextCompat.getSystemService(context, StorageManager.class);
+//
+//        String lastPathSegment = Objects.requireNonNull(uri.getLastPathSegment());
+//        String backupVolume    = lastPathSegment.replaceFirst(":.*", "");
+//        String backupName      = lastPathSegment.replaceFirst(".*:", "");
+//
+//        StorageManager      storageManager = ServiceUtil.getStorageManager(context);
+//        List<StorageVolume> storageVolumes = storageManager.getStorageVolumes();
+//        StorageVolume       storageVolume  = null;
+//
+//        for (StorageVolume volume : storageVolumes) {
+//            if (Objects.equals(volume.getUuid(), backupVolume)) {
+//                storageVolume = volume;
+//                break;
+//            }
+//        }
+//
+//        if (storageVolume == null) {
+//            return backupName;
+//        } else {
+//            return context.getString(R.string.StorageUtil__s_s, storageVolume.getDescription(context), backupName);
+//        }
+//    }
```

Comment:
I guess this method is a unnecessary leftover... Am I right?

Reply:
indeed. I will delete it